### PR TITLE
Add navigation bar spec for merchant user

### PR DESCRIPTION
**Pull request by Joshua-Tukman**
This finishes the functionality for navigation bar links specific to each type of user.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -21,9 +21,14 @@
...
         <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-        <%= link_to "Logout", "/logout"%>
+        <%= link_to "Logout", "/logout" %>
         <%= link_to "User Profile", "/default_user/profile"%>
 
+      <% elsif current_user.merchant? %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Logout", "/logout"%>
+        <%= link_to "Merchant Dashboard", "/merchant/dashboard"%>
+
```
Comment (Rostammahabadi) [inline app/views/layouts/application.html.erb:31]: This looks great

File: spec/features/merchant_user/navigation_bar_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+RSpec.describe 'As an merchant employee user' do
+    describe 'in the navigation bar' do
+        it 'I see links to home page, items index page, merchants index page, dashboard, logout, and profile page' do
+
+            employee = User.create(name: "Employee user",
+                                       address: "99 Working Hard Lane",
+                                       city: "Los Angeles",
+                                       state: "California",
+                                       zip: "90210",
```
Comment (JPuchalla) [inline spec/features/merchant_user/navigation_bar_spec.rb:11]: Solid tests.

File: spec/features/merchant_user/navigation_bar_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                expect(page).to have_link("All Merchants")
+                expect(page).to have_link("All Items")
+                expect(page).to have_link("Cart: 0")
+                expect(page).to have_link("Merchant Dashboard")
+                expect(page).to_not have_link("Login")
+                expect(page).to_not have_link("Register")
+            end
+        end
+    end
+end
```
Comment (melatran) [inline spec/features/merchant_user/navigation_bar_spec.rb:31]: Test looks great.

**Comment by Rostammahabadi** [comment]
approved!

File: config/routes.rb [hunk 4]
```diff
@@ -37,15 +37,6 @@
     get '/dashboard', to: 'dashboard#index'
   end
 
-  get "/profile", to: 'profile#index'
```
Comment (melatran) [inline config/routes.rb:40]: Thanks for catching the duplicate

**Pull request merged**